packages/router/test/router_preloader.spec.ts

```diff
@@ -196,6 +200,272 @@ describe('RouterPreloader', () => {
            })));
   });
 
+  describe('should support preloading stratages', () => {
+    let delayLoadUnPasued: BehaviorSubject<string[]>;
+    let delayLoadObserver$: Observable<string[]>;
+    let events: Array<RouteConfigLoadStart|RouteConfigLoadEnd>;
+
+    const subLoadChildrenSpy = jasmine.createSpy('submodule');
+    const lazyLoadChildrenSpy = jasmine.createSpy('lazymodule');
+    const buildPreloader = (
+        delayObserver$: Observable<string[]>,
+        ): typeof PreloadingStrategy => {
+      class DelayedPreLoad implements PreloadingStrategy {
+        preload(route: Route, fn: () => Observable<any>): Observable<any> {
+          const routeName =
+              route.loadChildren ? (route.loadChildren as jasmine.Spy).and.identity : 'noChildren';
+          return delayObserver$.pipe(
+              filter(unpauseList => unpauseList.indexOf(routeName) !== -1), take(1),
+              switchMap(() => {
+                return fn().pipe(
+                    catchError(() => of(null)),
+                );
+              }));
+        }
+      }
+      return DelayedPreLoad;
+    };
+
+    @NgModule({
+      declarations: [LazyLoadedCmp],
+    })
+    class SharedModule {
+    }
+
+    @NgModule({
+      imports: [
+        SharedModule, RouterModule.forChild([
+          {path: 'LoadedModule1', component: LazyLoadedCmp},
+          {path: 'sub', loadChildren: subLoadChildrenSpy}
+        ])
+      ]
+    })
+    class LoadedModule1 {
+    }
+
+    @NgModule({
+      imports:
+          [SharedModule, RouterModule.forChild([{path: 'LoadedModule2', component: LazyLoadedCmp}])]
+    })
+    class LoadedModule2 {
+    }
+
+    beforeEach(() => {
+      delayLoadUnPasued = new BehaviorSubject<string[]>([]);
+      delayLoadObserver$ = delayLoadUnPasued.asObservable();
+
+      subLoadChildrenSpy.calls.reset();
+      lazyLoadChildrenSpy.calls.reset();
+      const OurDelayedPreLoad = buildPreloader(delayLoadObserver$);
+      TestBed.configureTestingModule({
+        imports:
+            [RouterTestingModule.withRoutes([{path: 'lazy', loadChildren: lazyLoadChildrenSpy}])],
+        providers: [{provide: PreloadingStrategy, useClass: OurDelayedPreLoad}]
+      });
+      events = [];
+    });
+
+    it('without reloading loaded modules',
+       fakeAsync(inject([RouterPreloader, Router], (preloader: RouterPreloader, router: Router) => {
```

Comment:
```suggestion
       fakeAsync(() => {
       const preloader = TestBed.inject(RouterPreloader);
       const router = TestBed.inject(Router);
```
let's convert these to the more common `TestBed.inject` rather than the old-school `inject([]`) notation